src/api/TypedObserver.ts

```
@@ -0,0 +1,26 @@
+type Callback<T> = (...args: T[]) => void;
+
+class TypedObserver<T, FUNC extends Function = Callback<T>> {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)